assembl/graphql/schema.py

```diff
@@ -1,81 +1,439 @@
+    posts = graphene.List(PropositionPost)
+
+    def resolve_title(self, args, context, info):
+        title = resolve_langstring(self.title, args.get('lang'))
+        return title
+
+    def resolve_posts(self, args, context, info):
+        # TODO
+        return []
+
+
+class Thematic(SecureObjectType, SQLAlchemyObjectType):
+    class Meta:
+        model = models.Thematic
+        interfaces = (Node, )
+        only_fields = ('id', 'identifier')
+
+    title = graphene.String(lang=graphene.String())
+    description = graphene.String(lang=graphene.String())
+    questions = graphene.List(Question)
+    video = graphene.Field(Video, lang=graphene.String())
+    img_url = graphene.String()
+    num_posts = graphene.Int()
+    num_contributors = graphene.Int()
+
+    def resolve_title(self, args, context, info):
+        title = resolve_langstring(self.title, args.get('lang'))
+        return title
+
+    def resolve_description(self, args, context, info):
+        return resolve_langstring(self.description, args.get('lang'))
+
+    def resolve_questions(self, args, context, info):
+        return self.get_children()
+
+    def resolve_video(self, args, context, info):
+        title = resolve_langstring(self.video_title, args.get('lang'))
+        description = resolve_langstring(self.video_description, args.get('lang'))
+        return Video(
+            title=title,
+            description=description,
+            html_code=self.video_html_code,
+        )
+
+    def resolve_img_url(self, args, context, info):
+        # TODO imgUrl
+        return ""
 
 
 class Query(graphene.ObjectType):
-    node = URINode.Field()
-    agent_profiles = SQLAlchemyConnectionField(AgentProfile)
-    agent_profile = graphene.Field(AgentProfile)
-    user = graphene.Field(User)
-
-
-Schema = graphene.Schema(query=Query)
-
-"""
-query {
-    getUserAgent(id: $ID){
-        edge {
-            whatever
-        }
-    }
-}
-"""
-
-# this can execute:
-# schema.execute("query {node(id:\"local:AgentProfile/3\") {id ... on AgentProfile {name}} }")
-# schema.execute("query {agentProfiles {edges { node {id name}}} }")
-# This fails:
-# schema.execute("query {agentProfiles {edges { node {id name   ... on User {preferredEmail}}}} }")
-# schema.execute("query {agentProfiles(id:\"local:AgentProfile/3\") {id name} }")
+    node = Node.Field()
+    posts = SQLAlchemyConnectionField(PostConnection)#, idea_id=graphene.ID())
+    # ideas = SQLAlchemyConnectionField(Idea)
+    thematics = graphene.List(Thematic, identifier=graphene.String())
+    # agent_profiles = SQLAlchemyConnectionField(AgentProfile)
+
+    def resolve_ideas(self, args, context, info):
+        # TODO do we need to do a special query to only retrieve ideas
+        # associated to the RootIdea?
+        connection_type = info.return_type.graphene_type  # this is IdeaConnection
+        model = connection_type._meta.node._meta.model  # this is models.Idea
```

Comment:
Yeah if you want to retrieve what you wrote in the Meta class of the Idea objecttype, it's the only way. I wrote this code to maybe generalize it at one point so I can use it in other resolve functions.